File: gestionart_frontend_ruben_y_jessica/lib/data/models/SolicitudExclusiva.dart


import 'package:gestionart_frontend_ruben_y_jessica/data/enums/EstadoSolicitud.dart';

class Solicitudexclusiva {
  final int id;
  final int idComprador;
  final int idArticulo;
  final int idVendedor;
  final String mensage;
  final Estadosolicitud estado;
  final DateTime fecha;

  Solicitudexclusiva({required this.id, required this.idComprador, required this.idArticulo, required this.idVendedor, required this.mensage, required this.estado, required this.fecha});
  
  int getId() {
    return id;
  } 
  int getIdComprador() {
    return idComprador;
  }
  int getIdArticulo() {
    return idArticulo;
  }
  int getIdVendedor() {
    return idVendedor;
  }
  String getMensage() {
    return mensage;
  }
  Estadosolicitud getEstado() {
    return estado;
  }
  DateTime getFecha() {
    return fecha;
  }
  


  factory Solicitudexclusiva.fromJson(Map<String, dynamic> json) {
    return Solicitudexclusiva(
      id: json['id'] ?? 0,
      idComprador: json['idComprador'] ?? 0,
      idArticulo: json['idArticulo'] ?? 0,
      idVendedor: json['idVendedor'] ?? 0,
      mensage: json['mensage'] ?? '',
      estado: Estadosolicitud.values.firstWhere((e) => e.toString().split('.').last == 'Estadosolicitud.' + (json['estado'] ?? ''), orElse: () => Estadosolicitud.values.first),
      fecha: DateTime.parse(json['fecha'] ?? DateTime.now().toIso8601String())
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id ?? 0,
      'idComprador': idComprador ?? 0,
      'idArticulo': idArticulo ?? 0,
      'idVendedor': idVendedor ?? 0,
      'mensage': mensage ?? '',
      'estado': estado.toString().split('.').last ?? Estadosolicitud.values.first.toString().split('.').last,
      'fecha': fecha.toIso8601String() ?? DateTime.now().toIso8601String()
    };
  }
}

File: gestionart_backend_ruben_y_jessica/api/src/main/java/com/gestionart/api/domain/models/SolicitudExclusiva.java
package com.gestionart.api.domain.models;

import java.time.LocalDateTime;

import com.gestionart.api.domain.enums.EstadoSolicitud;

public class SolicitudExclusiva {
    private Long id;
    private Long idComprador;
    private Long idArticulo;
    private Long idVendedor;
    private String mensaje;
    private EstadoSolicitud estado;
    private LocalDateTime fecha;
    
    public SolicitudExclusiva() {
    }

    public SolicitudExclusiva(Long id, Long idComprador, Long idArticulo, Long idVendedor, String mensaje,EstadoSolicitud estado, LocalDateTime fecha) {
        this.id = id;
        this.idComprador = idComprador;
        this.idArticulo = idArticulo;
        this.idVendedor = idVendedor;
        this.mensaje = mensaje;
        this.estado = estado;
        this.fecha = fecha;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getIdComprador() {
        return idComprador;
    }

    public void setIdComprador(Long idComprador) {
        this.idComprador = idComprador;
    }

    public Long getIdArticulo() {
        return idArticulo;
    }

    public void setIdArticulo(Long idArticulo) {
        this.idArticulo = idArticulo;
    }

    public Long getIdVendedor() {
        return idVendedor;
    }

    public void setIdVendedor(Long idVendedor) {
        this.idVendedor = idVendedor;
    }

    public String getMensaje() {
        return mensaje;
    }

    public void setMensaje(String mensaje) {
        this.mensaje = mensaje;
    }

    public EstadoSolicitud getEstado() {
        return estado;
    }

    public void setEstado(EstadoSolicitud estado) {
        this.estado = estado;
    }

    public LocalDateTime getFecha() {
        return fecha;
    }

    public void setFecha(LocalDateTime fecha) {
        this.fecha = fecha;
    }

}
